library/core/src/stream/from_iter.rs

```diff
@@ -0,0 +1,53 @@
+use crate::pin::Pin;
+
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+/// A stream that was created from iterator.
+///
+/// This stream is created by the [`from_iter`] function.
+/// See it documentation for more.
+///
+/// [`from_iter`]: fn.from_iter.html
+#[unstable(feature = "stream_from_iter", issue = "81798")]
+#[derive(Clone, Debug)]
+pub struct FromIter<I> {
+    iter: I,
+}
+
+/// Converts an iterator into a stream.
+///
+/// # Examples
+///
+/// ```no_compile
+/// # std::thread::block_on(async {
+/// #
+/// use std::stream;
+///
+/// let mut s = stream::from_iter(vec![0, 1, 2, 3]);
+///
+/// assert_eq!(s.next().await, Some(0));
+/// assert_eq!(s.next().await, Some(1));
+/// assert_eq!(s.next().await, Some(2));
+/// assert_eq!(s.next().await, Some(3));
+/// assert_eq!(s.next().await, None);
```

Comment:
I understand it would be difficult to write examples without `next`, but I don't think it preferable to use functions that don't exist.